client/signup/main.jsx

```diff
@@ -161,6 +167,23 @@ class Signup extends React.Component {
 			providedDependencies = pick( queryObject, flow.providesDependenciesInQuery );
 		}
 
+		const searchParams = new URLSearchParams( window.location.search );
+		const isAddNewSiteFlow = searchParams.has( 'ref' );
+
+		if ( isAddNewSiteFlow ) {
+			clearSignupDestinationCookie();
+		}
+
+		const wasSignupCheckoutUnnloaded = getFromSessionStorage( 'signupCheckoutPageUnloaded' );
```

Comment:
Do you think it will be better to have functions explicitly get/set these values rather than relying on strings? For example - `get(set)SignupCheckoutPageUnloaded()`, `get(set)SignupCompleteSlug()`. The wrapper functions otherwise do not extend `sessionStorage` functionality.

Reply:
I'd like to second what @aneeshd16 proposed here, since it was actually what I meant when I left [this comment](https://github.com/Automattic/wp-calypso/pull/45040#discussion_r485482467) :)

Borrowing the terms from mathematics, the primary purpose of abstracting them into specific getter/setter here is to restrain the domains and the ranges of these functions so that they are unlikely to be misused. 

Also, I think `wasSignupCheckoutUnnloaded` has an redundant n here? i.e. Unnloaded -> Unloaded?

Reply:
Feedback addressed.